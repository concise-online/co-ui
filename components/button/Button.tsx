import React, { ButtonHTMLAttributes } from 'react';
import classNames from 'classnames';
import { getPrefixCls } from '../utils/config';

const prefixCls = getPrefixCls('btn');

interface Props extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'type'> {
  type?: 'primary' | 'success' | 'info' | 'warning' | 'danger';
  htmlType?: 'button' | 'submit' | 'reset';
  fill?: boolean;
}

function Button(props: Props) {
  const { children, className, htmlType, type, fill, ...btnProps } = props;
  const classes = classNames(prefixCls, className, {
    [`${prefixCls}-${type}`]: !!type,
    [`${prefixCls}-fill`]: fill,
  });

  return (
    <button
      type={htmlType}
      className={classes}
      {...btnProps}
    >
      <span>{children}</span>
    </button>
  );
}

export default Button;
