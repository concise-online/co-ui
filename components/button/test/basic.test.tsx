import React from 'react';
import { shallow, render } from 'enzyme';
import Button from '../index';

describe('button basic feature', () => {
  it('renders correctly', () => {
    const wrapper = render(<Button>Test</Button>);

    expect(wrapper).toMatchSnapshot();
  });

  it('click event correctly', () => {
    let num = 0;
    const wrapper = shallow(<Button onClick={() => num = 1}>Click Me</Button>);

    wrapper.simulate('click');
    expect(num).toBe(1);
  });

  it('button html type correctly', () => {
    const wrapper = shallow(<Button htmlType="button">Button</Button>);

    expect(wrapper.is('[type="button"]')).toEqual(true);
    wrapper.setProps({ htmlType: 'submit' });
    expect(wrapper.is('[type="submit"]')).toEqual(true);
    wrapper.setProps({ htmlType: 'reset' });
    expect(wrapper.is('[type="reset"]')).toEqual(true);
  });

  it ('button text correctly', () => {
    const wrapper = shallow(<Button>Hi, user</Button>);
    const wrapper1 = shallow(<Button>Hi, user{1 + 1}</Button>);

    expect(wrapper.children()).toHaveLength(1);
    expect(wrapper1.children()).toHaveLength(1);
  });
});
