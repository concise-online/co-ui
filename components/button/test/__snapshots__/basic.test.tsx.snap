// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button basic feature renders correctly 1`] = `
<button
  class="co-btn"
>
  <span>
    Test
  </span>
</button>
`;
